controllers/dscinitialization/auth.go

```diff
@@ -3,34 +3,74 @@ package dscinitialization
 import (
 	"context"
 
+	apiextv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	k8serr "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
 
+	dsciv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
 	serviceApi "github.com/opendatahub-io/opendatahub-operator/v2/apis/services/v1alpha1"
 	"github.com/opendatahub-io/opendatahub-operator/v2/controllers/components/dashboard"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster/gvk"
 )
 
-func (r *DSCInitializationReconciler) createAuth(ctx context.Context) error {
-	// Create Auth CR singleton
-	defaultAuth := client.Object(&serviceApi.Auth{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       serviceApi.AuthKind,
-			APIVersion: serviceApi.GroupVersion.String(),
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name: serviceApi.AuthInstanceName,
-		},
-		Spec: serviceApi.AuthSpec{
-			AdminGroups:   []string{dashboard.GetAdminGroup()},
-			AllowedGroups: []string{"system:authenticated"},
-		},
-	},
-	)
-	err := r.Create(ctx, defaultAuth)
+func (r *DSCInitializationReconciler) createAuth(ctx context.Context, dscInit *dsciv1.DSCInitialization) error {
+	log := logf.FromContext(ctx)
+	// check for the dashboardConfig kind.
+	// Once the groupsConfig entry in the dashboardConfig is removed this logic can be removed.
+	crd := &apiextv1.CustomResourceDefinition{}
+	err := r.Client.Get(ctx, client.ObjectKey{Name: "odhdashboardconfigs.opendatahub.io"}, crd)
+
+	// can't find the dashboard kind so create the default.
+	if err != nil {
+		err = r.Create(ctx, defaultAuth(true))
+		if err != nil && !k8serr.IsAlreadyExists(err) {
+			return err
+		}
+	}
+	// check if an instance exists.
+	odhObject := &unstructured.Unstructured{}
+	odhObject.SetGroupVersionKind(gvk.OdhDashboardConfig)
+	err = r.Client.Get(ctx, client.ObjectKey{
+		Name:      "odh-dashboard-config",
+		Namespace: dscInit.Spec.ApplicationsNamespace,
+	}, odhObject)
+	if err != nil {
+		log.Info("couldn't find odhDashboardConfig, creating auth with defaults.")
+	}
+	// if the kind exists and we have an odh-dashboard-config check for the groupsConfig and leave out the defaults if we find it.
+	if err == nil {
+		_, found, _ := unstructured.NestedStringMap(odhObject.Object, "spec", "groupsConfig")
+		if found {
+			err = r.Create(ctx, defaultAuth(false))
+			if err != nil && !k8serr.IsAlreadyExists(err) {
+				return err
+			}
+		}
+	}
+
+	err = r.Create(ctx, defaultAuth(true))
```

Comment:
probably it can be simplified with something like:

```go
useDefaults := true
if err != nil {
    _, ok, err := unstructured.NestedStringMap(odhObject.Object, "spec", "groupsConfig")
   if err != nil {
       return err
   } 
   
   useDefaults = !ok
}

err = r.Create(ctx, defaultAuth(useDefaults))
```

Reply:
:+1: